shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -810,13 +857,233 @@ bool OverrideGlobalPropertyFromIsolatedWorld(
+        "Failed to serialize function");
+    return v8::Local<v8::Value>();
+  }
+  v8::String::Utf8Value serialized_func_utf8(isolate, serialized_func);
+  std::string func_str(*serialized_func_utf8, serialized_func_utf8.length());
+
+  v8::Local<v8::Context> source_context = isolate->GetCurrentContext();
+  v8::Context::Scope source_scope(source_context);
+
+  // Get the target context
+  v8::MaybeLocal<v8::Context> maybe_target_context =
+      GetTargetContext(isolate, world_id);
+  v8::Local<v8::Context> target_context;
+  if (!maybe_target_context.ToLocal(&target_context)) {
+    isolate->ThrowException(v8::Exception::Error(gin::StringToV8(
+        isolate,
+        base::StringPrintf("Failed to get context for world %d", world_id))));
+    return v8::Local<v8::Value>();
+  }
+
+  // Compile the script
+  v8::MaybeLocal<v8::Script> maybe_compiled_script;
+  std::string error_message = "Unknown error during script compilation";
+  {
+    v8::Context::Scope target_scope(target_context);
+    v8::TryCatch try_catch(isolate);
+    std::string return_func_code = base::StringPrintf("(%s)", func_str.c_str());
+    maybe_compiled_script = v8::Script::Compile(
+        target_context, gin::StringToV8(isolate, return_func_code));
+    if (try_catch.HasCaught()) {
+      // Must throw outside of TryCatch scope
+      v8::String::Utf8Value error(isolate, try_catch.Exception());
+      error_message =
+          *error ? *error : "Unknown error during script compilation";
+    }
+  }
+  v8::Local<v8::Script> compiled_script;
+  if (!maybe_compiled_script.ToLocal(&compiled_script)) {
+    isolate->ThrowException(
+        v8::Exception::Error(gin::StringToV8(isolate, error_message)));
+    return v8::Local<v8::Value>();
+  }
+
+  // Run the script
+  v8::MaybeLocal<v8::Value> maybe_script_result;
+  {
+    v8::Context::Scope target_scope(target_context);
+    v8::TryCatch try_catch(isolate);
+    maybe_script_result = compiled_script->Run(target_context);
+    if (try_catch.HasCaught()) {
+      // Must throw outside of TryCatch scope
+      v8::String::Utf8Value error(isolate, try_catch.Exception());
+      error_message = *error ? *error : "Unknown error during script execution";
+    }
+  }
+  v8::Local<v8::Value> script_result;
+  if (!maybe_script_result.ToLocal(&script_result)) {
+    isolate->ThrowException(
+        v8::Exception::Error(gin::StringToV8(isolate, error_message)));
+    return v8::Local<v8::Value>();
+  }
+
+  // Get copied function from the script result
+  if (!script_result->IsFunction()) {
+    isolate->ThrowException(v8::Exception::Error(
+        gin::StringToV8(isolate, "Unknown error retrieving function")));
+    return v8::Local<v8::Value>();
+  }
+  v8::Local<v8::Function> copied_func = script_result.As<v8::Function>();
+
+  // Proxy args to be passed into copied function
+  std::vector<v8::Local<v8::Value>> proxied_args;
+  {
+    bool support_dynamic_properties = false;
+    context_bridge::ObjectCache object_cache;
```

Comment:
It's weird calling `PassValueToOtherContext` multiple times with the same object cache, it works but it's not intended. Is there a DX reason we didn't just call `PassValueToOtherContext` on the entire array?

Reply:
The argument iteration provides visibility into which exact argument would throw when passing them into the other context. The error messages are formatted to include the argument index.

Open to creating an object cache per-argument, I can lean on your input here as the original code author. I saw that it was possible and did it 😄

Reply:
I noticed ProxyFunctionWrapper is using a similar argument iteration and uses a shared object cache. I was probably referencing this when implementing.
https://github.com/electron/electron/blob/8f00cc9c0e78b95e4de14b9b9e1a7df20935e948/shell/renderer/api/electron_api_context_bridge.cc#L465-L481

Reply:
See https://github.com/electron/electron/pull/44411#discussion_r1864078646 for how I addressed this.